performance_manager/lib/gtfs_static_table.py

```diff
@@ -0,0 +1,281 @@
+    dataframe into StaticTableDetails objects
+    """
+    for table in static_tables:
+        paths_to_load = get_static_parquet_paths(
+            table.table_name, feed_info_path
+        )
+        table.data_table = read_parquet(
+            paths_to_load, columns=table.columns_to_pull
+        )
+        assert table.data_table.shape[0] > 0
+
+
+def transform_data_tables(static_tables: List[StaticTableDetails]) -> None:
+    """
+    transform static gtfs schedule dataframe objects as required
+    """
+    for table in static_tables:
+        table.data_table = table.data_table.drop_duplicates()
+
+        if table.int64_cols is not None:
+            for col in table.int64_cols:
+                table.data_table[col] = pandas.to_numeric(
+                    table.data_table[col]
+                ).astype("Int64")
+
+        if table.bool_cols is not None:
+            for col in table.bool_cols:
+                table.data_table[col] = numpy.where(
+                    table.data_table[col] == 1, True, False
+                ).astype(numpy.bool8)
+
+        if table.time_to_seconds_cols is not None:
+            for col in table.time_to_seconds_cols:
+                table.data_table[col] = (
+                    table.data_table[col]
+                    .apply(start_time_to_seconds)
+                    .astype("Int64")
+                )
+
+
+def insert_data_tables(
+    static_tables: List[StaticTableDetails],
+    session: sa.orm.session.sessionmaker,
+) -> None:
+    """
+    insert static gtfs data tables into rds tables
+    """
+    for table in static_tables:
+        with session.begin() as cursor:  # type: ignore
+            cursor.execute(
+                sa.insert(table.insert_table),
+                table.data_table.to_dict(orient="records"),
+            )
+
+
+def process_static_tables(sql_session: sessionmaker) -> None:
+    """
+    process gtfs static table files from metadataLog table
+    """
+
+    # pull list of objects that need processing from metadata table
+    paths_to_load = get_unprocessed_files("FEED_INFO", sql_session)
+
+    for folder, folder_data in paths_to_load.items():
+        ids = folder_data["ids"]
+        folder = str(folder)
+
+        static_tables = get_table_objects()
+
+        try:
+            load_parquet_files(static_tables, folder)
+            transform_data_tables(static_tables)
+            insert_data_tables(static_tables, sql_session)
+        except Exception as e:
+            logging.info("Error Processing Static GTFS Schedules")
+            logging.exception(e)
+        else:
+            update_md_log = (
+                sa.update(MetadataLog.__table__)
+                .where(MetadataLog.pk_id.in_(ids))
+                .values(processed=1)
+            )
+            with sql_session.begin() as cursor:  # type: ignore
+                cursor.execute(update_md_log)
+
+        # delete static_tables to ensure memory freed
+        del static_tables
```

Comment:
Make sure list of objects is deleted

Reply:
i think we can just let this get GC'ed as it falls out of scope?